cmd/launch_producer/main.cpp

```diff
@@ -159,8 +162,43 @@ FN_PTR loadLibrary(const std::string& path, const char* lib) {
   return startFunc;
 }
 
+void killExistingProcess() {
+  int fd = open(kPidFileName, O_RDONLY);
+  if (fd == -1) {
+    return;
+  }
+  char pidString[10];
+  if (read(fd, pidString, 10) > 0) {
+    int pid = -1;
+    sscanf(pidString, "%d", &pid);
+    if (pid > 0) {
+      kill(pid, SIGINT);
+    }
+  }
+  close(fd);
+}
+
+bool writeToPidFile() {
+  killExistingProcess();
```

Comment:
Add a comment along these lines:
```
If a previous producer has died without cleaning up its pidfile, here we kill a PID that may be related to
another process. This is a risk we take, it would be rare for a previous PID to be reused, and in the
worst case we kill a non-critical application (core services are not killable that easily).
```